LGAlertView/LGAlertViewController.m

```diff
@@ -52,14 +52,29 @@ - (nonnull instancetype)initWithAlertView:(nonnull LGAlertView *)alertView view:
 }
 
 - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
-    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    if (coordinator) {
+        [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    }
 
-    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.0 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-        [UIView animateWithDuration:coordinator.transitionDuration animations:^{
-            [self setNeedsStatusBarAppearanceUpdate];
-            [self.alertView layoutValidateWithSize:size];
+    @try {
+        [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+            __weak typeof(self) weakSelf = self;
+            [UIView animateWithDuration:0.5 animations:^{
+                if (weakSelf) {
```

Comment:
Why not making it strong after check?
```objective-c
if (!weakSelf) {
  return;
}
__string typeof(weakSelf) strongSelf = weakSelf;
// use strongSelf below
```

Reply:
I think it allowed, but can't understand why we need it?

Reply:
Well, the general idea behind `weak self` and `string self` is when we are using block in different thread it might be the case when the `self` itself has been destroyed before block execution started, that't why we make self **weak** so to check if self is still alive like `if (weakSelf)`, but after that, in the process of block execution in the different thread `self` still may be destroyed in another thread and as `weak self` doesn't increase referense count and doesn't own the object it also can be destroyed together with self in another thread, that's why it is ususally necessery to wrap `weak self` into `strong self` after the check that it exists.